### DISCUSSION TITLE
How to start contributing to open source projects?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#18336] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi, im a high school student who just started learning to coding. I am keen to build my own projects and contribute to open source project. Are there any advice for new comers like me?

Thanks

**Comment by DavidYang2149** [comment]
Hello.
You can create public repository!
and fill up your code or document(for other people).
Good luck! 😄 

[Also see...](https://opensource.com/resources/what-open-source)

**Reply by ghost** [reply]
Thanks!

**Comment by Nishant1500** [comment]
> Hello. You can create public repository! and fill up your code or document(for other people). Good luck! 😄
> 
> [Also see...](https://opensource.com/resources/what-open-source)

Yep as they said ^
Guide for new comers -> [**`Introduction to GitHub`**](https://lab.github.com/githubtraining/introduction-to-github)

**Reply by ghost** [reply]
Will cehck it out !

**Comment by melvincwng** [comment, marked as answer]
Hi @AbhisekChakra90, first of all, let me welcome you to Github Community!

For starters, since you are still in high school and started to learn how to code, my suggestion would be to learn the basics of frontend development first - i.e HTML5, CSS3, JavaScript. There are a couple of great courses from Youtube, Codecademy, and freeCodeCamp that you can utilize to get started!

Once you have learned the basics of frontend, you can try learning some backend-related technologies - e.g. Node.js/Express.js and MongoDB/MySQL databases. You can even experiment with learning backend with Python & Flask/Django. Again, there are many free and useful resources on the web which you can utilize to help in learning.

For personal projects, as a start, my advice is to start small. Perhaps you can find a pain point or a problem that you are trying to solve in your daily life. For example - need an app to track what are the tasks needed to be done for the day, then try creating a 'To-Do List app. Love to game? - then build a simple game app. The possibilities are endless ;)

For open-source projects, as a starter - check out https://github.com/firstcontributions/first-contributions. You simply need to edit and add your name as a contributor there to get a taste of contributing to open-source projects.

The key is to start small, stay consistent, and continuously learn (plus enjoy the process of being a software developer!). Eventually, as you gain proficiency in your technical skills, you can then explore other languages or even delve into Computer Science when you go to university in the future.

Hope this helps you out, and all the best in your journey!

**Reply by ghost** [reply]
Thank you for detailed response! I am learning HTML CSS and some javacsript now, it is fun learning new stuff so far.

**Comment by mequann** [comment]
I have worked on MERN Stack And need to contribute for projects please let me know  how do so please?
